securesystemslib/interface.py

```diff
@@ -277,37 +303,51 @@ def import_rsa_privatekey_from_file(filepath, password=None,
   # Is 'scheme' properly formatted?
   securesystemslib.formats.RSA_SCHEME_SCHEMA.check_match(scheme)
 
-  # If the caller does not provide a password argument, prompt for one.
-  # Password confirmation disabled here, which should ideally happen only
-  # when creating encrypted key files (i.e., improve usability).
-  if password is None: # pragma: no cover
+  if password and prompt:
+    raise ValueError("Passing 'password' and 'prompt' True is not allowed.")
+
+  # If 'password' was passed check format and that it is not empty.
+  if password is not None:
+    securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
 
+    # TODO: PASSWORD_SCHEMA should be securesystemslib.schema.AnyString(min=1)
+    if not len(password):
+      raise ValueError('Password must be 1 or more characters')
+
+  elif prompt:
+    # Password confirmation disabled here, which should ideally happen only
+    # when creating encrypted key files (i.e., improve usability).
     # It is safe to specify the full path of 'filepath' in the prompt and not
     # worry about leaking sensitive information about the key's location.
     # However, care should be taken when including the full path in exceptions
     # and log files.
```

Comment:
AFAIK, Trishank and Vlad made this decision, because in order for the prompt to show the full path, the full path must be known to the caller anyway, whereas an exception/log might reveal that information to someone who normally does not know it.

Reply:
> because in order for the prompt to show the full path, the full path must be known to the caller anyway, whereas an exception/log might reveal that information to someone who normally does not know it.

That's correct.  Revealing the full path via an exception or log file should be avoided elsewhere. 

We want to avoid vulnerabilities like [Information exposure through an error message](https://cwe.mitre.org/data/definitions/209.html) and [full path disclosure](https://www.owasp.org/index.php/Full_Path_Disclosure).